components/drafts/draft_title/draft_title.tsx

```diff
@@ -0,0 +1,143 @@
+    type: 'channel' | 'thread';
+}
+
+function DraftTitle({
+    channel,
+    channelName,
+    membersCount,
+    selfDraft,
+    teammate,
+    teammateId,
+    type,
+}: Props) {
+    const dispatch = useDispatch();
+
+    useEffect(() => {
+        if (!teammate?.id && teammateId) {
+            dispatch(getMissingProfilesByIds([teammateId]));
+        }
+    }, [teammate?.id, teammateId]);
+
+    let you = null;
+    if (selfDraft) {
+        you = (
+            <FormattedMessage
+                id='drafts.draft_title.you'
+                defaultMessage={'(you)'}
+            />
+        );
+    }
+
+    let icon = <i className='icon icon-globe'/>;
+
+    if (channel.type === Constants.DM_CHANNEL) {
+        icon = (
+            <Avatar
+                size='xs'
+                username={teammate?.username}
+                url={imageURLForUser(teammate?.id, teammate?.last_picture_update)}
+                className='DraftTitle__avatar'
+            />
+        );
+    }
+
+    if (channel.type === Constants.GM_CHANNEL) {
+        icon = (
+            <div className='DraftTitle__group-icon'>
+                {membersCount}
+            </div>
+        );
+    }
+
+    if (type === 'thread') {
+        if (
+            channel.type !== Constants.GM_CHANNEL &&
+            channel.type !== Constants.DM_CHANNEL
+        ) {
+            return (
+                <>
+                    <FormattedMessage
+                        id='drafts.draft_title.channel_thread'
+                        defaultMessage={'Thread in: {icon} {channelName}'}
+                        values={{
+                            icon,
+                            channelName,
+                        }}
+                    />
+                    &nbsp;{you}
+                </>
+            );
+        }
+
+        return (
+            <>
+                <FormattedMessage
+                    id='drafts.draft_title.direct_thread'
+                    defaultMessage={'Thread to: {icon} {channelName}'}
+                    values={{
+                        icon,
+                        channelName,
+                    }}
+                />
+                &nbsp;{you}
+            </>
+        );
+    }
+
+    if (
+        channel.type !== Constants.GM_CHANNEL &&
+        channel.type !== Constants.DM_CHANNEL
+    ) {
+        return (
+            <FormattedMessage
+                id='drafts.draft_title.channel'
+                defaultMessage={'In: {icon} {channelName}'}
+                values={{
+                    icon,
+                    channelName,
+                }}
+            />
+        );
+    }
+
+    return (
+        <>
+            <FormattedMessage
+                id='drafts.draft_title.direct_channel'
+                defaultMessage={'To: {icon} {channelName}'}
+                values={{
+                    icon,
+                    channelName,
+                }}
+            />
+            &nbsp;{you}
```

Comment:
Is this space always needed? I have seen automated tests failing because of extra spaces where they are not needed, so just making sure.